### PR TITLE
Start using a non-global reactor.

### PR DESCRIPTION
**Pull request by tomprince**
It is much easier to test code that doesn't use the global reactor.

Change `BuildMaster` to take an optional reactor as a first step.

### CODE REVIEW

File: master/buildbot/test/util/integration.py [hunk 1]
```diff
@@ -81,6 +81,14 @@ def setupConfig(self, configFunc):
                         configFunc=configFunc,
                         proto=proto,
                         workerclass=workerclass))
+
+        # mock reactor.stop (which trial *really* doesn't
```
Comment (tardyp) [inline master/buildbot/test/util/integration.py]: shall we factorize those lines?

**Comment by tardyp** [comment]
I think we should have a buildbot.unit.fake.reactor.Reactor class. good change apart from that!

**Comment by tomprince** [comment]
> I think we should have a buildbot.unit.fake.reactor.Reactor class.

I've started on one.

File: master/buildbot/test/util/integration.py (hunk 1 above)
Comment (tomprince) [inline master/buildbot/test/util/integration.py]: I'm inclined to wait until there is a fake reactor to do that.

Comment (tardyp) [inline master/buildbot/test/util/integration.py]: so this is for a followup PR?

Comment (tomprince) [inline master/buildbot/test/util/integration.py]: I plan to introduce a better fake reactor, and I think that any factorization of this code should use that fake reactor. I can't commit to adapting all the uses of this to use a fake reactor in timely fashion, though.

**Comment by djmitche** [comment]
+1 from me -- good idea

**Pull request merged**